### PR TITLE
Wrap `Frame`, `Card` and `Actions` patterns with components

### PR DESCRIPTION
**Pull request by lyzadanger**
This PR:

* Provides `Frame`, `Card` and `Actions` components. I am calling these components _content components_ for the time being: they are components that provide no real logic, but apply design patterns for authors, consistently.
* Adds a `classes` prop to each of the content components, as well as the pre-existing `TextInput` and `TextInputWithButton` components. This allows authors control over overriding styling.
* Renames all things that were previously called `molecules` to `content`. This feels like a more intuitive name for this group of patterns and components. I imagine `organisms` will be renamed shortly, as well. The atomic nomenclature is helpful to a point, but I'd like to set it aside when it feels cumbersome or arbitrary.

Screen shot of the "Content" components page in the pattern library after these changes. Note the "Content" patterns and components sections (they correspond to each other).

image

### Second-pass naming

Based on feedback to the initial _content_ naming, the updated branch uses _containers_ instead of the _content_ term used above:

image

Fixes https://github.com/hypothesis/frontend-shared/issues/115
Fixes https://github.com/hypothesis/frontend-shared/issues/121

### CODE REVIEW

File: src/components/test/content-test.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+    assert.isTrue(wrapper.find('div').first().hasClass('Hyp-Card'));
+  });
+
+  it('applies extra classes', () => {
+    const wrapper = createComponent({ classes: 'foo bar' });
+
+    assert.isTrue(wrapper.find('div').first().hasClass('Hyp-Card'));
+    assert.isTrue(wrapper.find('div').first().hasClass('foo'));
+    assert.isTrue(wrapper.find('div').first().hasClass('bar'));
```
Comment (LMS007) [inline src/components/test/content-test.js:47]: Same pattern here..
`assert.exists(...)`

File: src/components/test/content-test.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  it('renders children inside of a div with appropriate classnames', () => {
+    const wrapper = createComponent();
+
+    assert.isTrue(wrapper.find('div').first().hasClass('Hyp-Actions--row'));
+  });
+
+  it('applies extra classes', () => {
+    const wrapper = createComponent({ classes: 'foo bar' });
+
+    assert.isTrue(wrapper.find('div').first().hasClass('Hyp-Actions--row'));
```
Comment (LMS007) [inline src/components/test/content-test.js:68]: Same pattern here..
`assert.exists(...)`

**Review by LMS007 — APPROVED** [review]
I concur with these changes. I think this is moving in the right direction. I did find one reference to`molecules` in the [README](https://github.com/hypothesis/frontend-shared/blob/content-components/styles/README.md#q-how-much-do-i-need-to-know-about-atomic-design-to-get-my-job-done-here) we probably want to rename or remove. 

I just had some small suggestions on the tests cases to make them a bit more terse, but otherwise solid!

I do feel the name "Content Components" is not quite right probably because it feels too generic at least to me, and the word content is overloaded all over the place.

If I may make a name suggestion, how about "Shell Components". Its very specific and concisely explains its purpose.

Thoughts?

File: src/components/content.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
+/**
+ * Components that help to lay out and style content
+ */
+
+import classnames from 'classnames';
+
+/**
+ * @typedef {import('preact').ComponentChildren} Children
+ * @typedef MoleculeProps
```
[the remark this answers is not in the document]
Comment (lyzadanger) [reply src/components/content.js:9]: No, you're right, I overlooked this change. Thanks!

File: src/components/test/TextInput-test.js [hunk 4]
```diff
@@ -23,12 +23,20 @@ describe('TextInput', () => {
 
     assert.isTrue(wrapper.find('input').hasClass('is-error'));
   });
+
+  it('applies extra classes', () => {
+    const wrapper = createComponent({ classes: 'foo bar' });
+
+    assert.isTrue(wrapper.find('input').first().hasClass('Hyp-TextInput'));
```
[the remark this answers is not in the document]
Comment (lyzadanger) [reply src/components/test/TextInput-test.js:30]: Likely a c-n-p artifact here. I'll address.

**Comment by lyzadanger** [comment]
@LMS007 

> I do feel the name "Content Components" is not quite right probably because it feels too generic at least to me, and the word content is overloaded all over the place.

You know, I'm inclined to agree, on reflection. The good news is that whatever we choose is package-internal (consumers just import components directly, with no knowledge of how they're organized internally), so we have some flexibility. What do you think about _container_? "Container components". You put things (content, etc.) in them.

File: src/components/content.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
+/**
+ * Components that help to lay out and style content
+ */
+
+import classnames from 'classnames';
+
+/**
+ * @typedef {import('preact').ComponentChildren} Children
```
[the remark this answers is not in the document]
Comment (lyzadanger) [reply src/components/content.js:8]: That's a very reasonable request. Adding an extra line break here.

File: src/components/test/content-test.js [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+
+    assert.isTrue(wrapper.find('div').first().hasClass('Hyp-Frame'));
+  });
+
+  it('applies extra classes', () => {
+    const wrapper = createComponent({ classes: 'foo bar' });
+
+    assert.isTrue(wrapper.find('div').first().hasClass('Hyp-Frame'));
+    assert.isTrue(wrapper.find('div').first().hasClass('foo'));
+    assert.isTrue(wrapper.find('div').first().hasClass('bar'));
```
[the remark this answers is not in the document]
Comment (lyzadanger) [reply src/components/test/content-test.js:24]: OK, that `assert.exists` is just magicsauce. I've got a little digging around to do to understand where it gets the savvy to know where to "look" for that `div`. Enzyme magic?

**Comment by lyzadanger** [comment]
@LMS007 I have rebuilt the branch (though trying to adhere to the same structure of the original three commits) because of the amount of file-renaming involved here. In this second pass:

* Using naming of _containers_ instead of _content_ throughout
* Tests are simplified (nice suggestion!) by using `assert.exists`

**Review by LMS007 — APPROVED** [review]
Container Components is solid

GTG!

**Pull request merged**